File: lazy-professor/lazy.pl
open $fh, '<', './test.txt';

my @answers;
while (<$fh>) {
  chomp;
  my $i;
  $answers[$i++]{$_}++ for split ''
}

for (@answers) {
  my $most_frequent = [];
  while (my ($answer, $count) = each %$_) {
    $most_frequent ||= [$answer, $count];
    $most_frequent = [$answer, $count] if $count > $most_frequent->[1]
  }
  print $most_frequent->[0]
}

print "\n"

File: lazy-professor/lazy.rb
puts File.open('./test.txt')
  .map { |x| x.split('') }
  .reduce([]) { |a,cs| cs.each_with_index { |c,i| a[i] ||= {}; a[i][c] ||= 0; a[i][c] += 1 }; a }
  .map { |cs| cs.max_by { |k,v| v } }.map(&:first).join('')
